test/test_api.py

```diff
@@ -0,0 +1,235 @@
+
+    db.session.add(profile)
+    db.session.commit()
+
+    response = test_client.get('/profiles/a1b2c3d4')
+
+    assert response.status_code == 200
+    assert response.headers.get('Content-Type') == 'application/vnd.elife.profile+json;version=1'
+    assert json.loads(response.data.decode('UTF-8'))['id'] == 'a1b2c3d4'
+
+
+def test_profile_not_found(test_client: FlaskClient) -> None:
+    response = test_client.get('/profiles/foo')
+
+    assert response.status_code == 404
+    assert response.headers.get('Content-Type') == 'application/problem+json'
+
+
+def test_get_profile_response_contains_email_addresses(test_client: FlaskClient) -> None:
+    profile = Profile('a1b2c3d4', Name('Foo Bar'), '0000-0002-1825-0097')
+    profile.add_email_address('1@example.com')
+    profile.add_email_address('2@example.com')
+
+    db.session.add(profile)
+    db.session.commit()
+
+    response = test_client.get('/profiles/a1b2c3d4')
+    response_data = json.loads(response.data.decode('UTF-8'))
+
+    assert response.status_code == 200
+    assert response.headers.get('Content-Type') == 'application/vnd.elife.profile+json;version=1'
+    assert response_data['emailAddresses'] == ['1@example.com', '2@example.com']
+
+
+def test_get_profile_response_contains_affiliations(test_client: FlaskClient) -> None:
+    address = Address(country=countries.get('gb'), city='City', region='Region')
+    affiliation = Affiliation('1', address=address, organisation='Org',
+                              department='Dep', starts=datetime.now())
+
+    profile = Profile('a1b2c3d4', Name('Foo Bar'), '0000-0002-1825-0097')
+
+    db.session.add(profile)
+    profile.add_affiliation(affiliation)
+
+    db.session.commit()
+
+    response = test_client.get('/profiles/a1b2c3d4')
+    response_data = json.loads(response.data.decode('UTF-8'))
+
+    assert response.status_code == 200
+    assert response.headers.get('Content-Type') == 'application/vnd.elife.profile+json;version=1'
+    assert len(response_data['affiliations']) == 1
+    assert response_data['affiliations'][0]['name'] == ['Dep', 'Org']
+    assert response_data['affiliations'][0]['address']['formatted'] == ['City', 'Region', 'GB']
+
+
+def test_get_profile_response_against_json_schema(test_client: FlaskClient) -> None:
+    address = Address(country=countries.get('gb'), city='City', region='Region')
+    affiliation = Affiliation('1', address=address, organisation='Org',
+                              department='Dep', starts=datetime.now())
+
+    profile = Profile('a1b2c3d4', Name('Foo Bar'), '0000-0002-1825-0097')
+
+    db.session.add(profile)
+    profile.add_affiliation(affiliation)
+
+    db.session.commit()
+
+    response = test_client.get('/profiles/a1b2c3d4')
+    response_data = json.loads(response.data.decode('UTF-8'))
+
+    assert validate_json(response_data, schema_name='profile.v1') is True
+
+
+def test_get_list_of_profiles_response_against_json_schema(test_client: FlaskClient) -> None:
```

Comment:
Ideally all the above tests would validate instead of having separate tests.

Reply:
Ok done.